src/tss2-esys/esys_iutil.c

```diff
@@ -1295,6 +1295,18 @@ iesys_gen_auths(ESYS_CONTEXT * esys_context,
                                     &encryptNonce);
     return_if_error(r, "More than one crypt session");
 
+    /*
+     * TPM2.0 Architecture 19.6.5 Note 7
+     *
+     * If the same session (not the first session) is used for decrypt and
+     * encrypt, its nonceTPM is only used once. If different sessions are used
+     * for decrypt and encrypt, both nonceTPMs are included
+     */
+    if (decryptNonceIdx == encryptNonceIdx) {
```

Comment:
So it should perhaps be:
```
if (decryptNonceIdx > 0 && decryptNonceIdx == encryptNonceIdx) {
        decryptNonceIdx = 0;
}
```

Reply:
Yes I agree, to this solution. The encryptNonce should not be set to 0.

Reply:
I would suggest:
```
if (encryptNonceIdx > 0 && decryptNonceIdx == encryptNonceIdx) {
        encryptNonceIdx = 0;
}